app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/BusinessIdentificationController.scala

```diff
@@ -528,4 +525,14 @@ class BusinessIdentificationController @Inject()(
   private def cacheInitialDetailsAndRedirect(initialDetails: InitialDetails)(call: => Call)(
     implicit hc: HeaderCarrier): Future[Call] =
     sessionStoreService.cacheInitialDetails(initialDetails).map(_ => call)
+
+  def isValidBusinessName(details: InitialDetails): Boolean = initialDetailsValidator.validate(details) match {
+    case Failure(responses) if responses.contains(InvalidBusinessName) => true
+    case _                                                             => false
+  }
+
+  def isValidEmail(details: InitialDetails): Boolean = initialDetailsValidator.validate(details) match {
+    case Failure(responses) if responses.contains(InvalidEmail) => true
```

Comment:
The true & false cases need to be flipped both in this `isValidEmail` and in `isValidBusinessName`

Reply:
Or the functions renamed to `isInvalidEmail` and `isInvalidBusinessName`